From 2521548af4d3ba27ea1bd9d0bcb06236f55d89e8 Mon Sep 17 00:00:00 2001
From: Stan Angeloff <stanimir@angeloff.name>
Date: Wed, 13 Aug 2025 14:25:35 +0300
Subject: [PATCH 2/2] feat: replace `dotool` with `wl-copy`.

---
 old.go | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/old.go b/old.go
index 72e35f5..91920f2 100644
--- a/old.go
+++ b/old.go
@@ -142,29 +142,29 @@ Listen:
 			continue Listen
 		}
 
 		log.Println("main: transcribed text: ", resp.Text)
 
-		dotool := exec.Command("dotool")
-		stdin, err := dotool.StdinPipe()
+		wlcopy := exec.Command("wl-copy")
+		stdin, err := wlcopy.StdinPipe()
 		if err != nil {
-			log.Println("main: ", fmt.Errorf("dotool stdin pipe: %w", err))
+			log.Println("main: ", fmt.Errorf("wl-copy stdin pipe: %w", err))
 		}
-		dotool.Stderr = os.Stderr
-		if err := dotool.Start(); err != nil {
-			log.Println("main: ", fmt.Errorf("dotool start: %w", err))
+		wlcopy.Stderr = os.Stderr
+		if err := wlcopy.Start(); err != nil {
+			log.Println("main: ", fmt.Errorf("wl-copy start: %w", err))
 		}
 
-		_, err = io.WriteString(stdin, fmt.Sprintf("type %s", resp.Text))
+		_, err = io.WriteString(stdin, resp.Text)
 		if err != nil {
-			log.Println("main: ", fmt.Errorf("dotool stdin WriteString: %w", err))
+			log.Println("main: ", fmt.Errorf("wl-copy stdin WriteString: %w", err))
 		}
 
 		if err := stdin.Close(); err != nil {
-			log.Println("main: close dotool stdin: ", err)
+			log.Println("main: close wl-copy stdin: ", err)
 		}
 
-		if err := dotool.Wait(); err != nil {
-			log.Println("main: dotool wait: ", err)
+		if err := wlcopy.Wait(); err != nil {
+			log.Println("main: wl-copy wait: ", err)
 		}
 	}
 }
-- 
2.50.1

